packages/ember/src/components/form/phone-field.gts

```diff
@@ -0,0 +1,120 @@
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { runTask } from 'ember-lifeline';
+
+import TextField from './text-field.gts';
+import onInsert from '../../modifiers/did-insert.ts';
+
+import type { Optional } from '../../types.d.ts';
+
+export default class PhoneField extends TextField {
+  @tracked
+  inputElement: Optional<HTMLInputElement> = null;
+
+  @action
+  onInsert(element: HTMLElement) {
+    this.inputElement = element as HTMLInputElement;
+  }
+
+  @action
+  change(evt: Event) {
+    const inputEvent = evt as InputEvent;
+    const target = evt.target as HTMLInputElement;
+    const newValue = target?.value;
+    const currentValue = this.displayValue;
+    const cursorPosition = target?.selectionStart ?? -1;
+    const isBackspace = inputEvent.inputType === 'deleteContentBackward';
+    const isDelete = inputEvent.inputType === 'deleteContentForward';
+
+    let unformattedValue = newValue.replace(/\D/g, '');
+    let characterUnderCursor = currentValue[cursorPosition ?? 0] ?? '';
+    const isNonDigitCharacter = /\D/.test(characterUnderCursor);
+    const isDeletingSpecialCharacter =
+      cursorPosition >= 0 && (isBackspace || isDelete) && isNonDigitCharacter;
+    if (isDeletingSpecialCharacter) {
+      if (isBackspace) {
+        let newCursorPosition = cursorPosition - 1;
+        if (/\D/.test(currentValue[newCursorPosition ?? 0] ?? '')) {
+          newCursorPosition--;
+        }
+        this.inputElement?.setSelectionRange(
+          newCursorPosition,
+          newCursorPosition,
+        );
+      }
+      const beforeCursor = newValue
+        .substring(0, cursorPosition)
+        .replace(/\D/g, '');
+      const afterCursor = newValue.substring(cursorPosition).replace(/\D/g, '');
+      if (isBackspace) {
+        unformattedValue = `${beforeCursor.slice(0, -1)}${afterCursor}`;
```

Comment:
I think the template literals make this a little hard to read, but it can be written as simple concatenation:
```suggestion
        unformattedValue = beforeCursor.slice(0, -1) + afterCursor;
```